pkg/config/config.go

```diff
@@ -205,6 +225,51 @@ func (c *Config) getPaths(dir string, client *github.Client) ([]*IBCData, error)
 	return ibcs, nil
 }
 
+func (c *Config) Validate() error {
+	validate := validator.New(validator.WithRequiredStructEnabled())
+
+	// register custom validation for http url as expected by go relayer i.e.
+	// http_url must have port defined.
+	// https://github.com/cosmos/relayer/blob/259b1278264180a2aefc2085f1b55753849c4815/cregistry/chain_info.go#L115
+	err := validate.RegisterValidation("has_port", func(fl validator.FieldLevel) bool {
+		val := fl.Field().String()
+		valArr := strings.Split(val, ":")
```

Comment:
I think it is better to use `Parse` method from `net/url` package and later `Port` method on parsed url to get the port. Your code doesn't fail validation for an url provided like `https://rpc.constantine.archway.tech?param:12`